R/dbi-methods.R

```diff
@@ -183,16 +191,28 @@ setMethod("dbSendQuery", c("SQLServerConnection", "character"),
 #' @export
 
 setMethod("dbSendStatement", c("SQLServerConnection", "character"),
-  function(conn, statement, params = NULL, ...) {
+  function(conn, statement, params = NULL, ..., batch = FALSE) {
     assertthat::assert_that(assertthat::is.string(statement),
       is.null(params) || is.list(params))
-    if(!is.null(params)) {
-      s <- create_prepared_statement(conn, statement)
-      catch_exception(s, "Unable to create prepared statement ", statement)
-      on.exit(close_statement(s))
-      ps_bind_all(params, s)
-      res <- execute_update(s, check = TRUE)
+    if (length(params)) {
```

Comment:
`length(list())` is effectively false, whereas `!is.null(list())` is true. Just guarding against user-error.